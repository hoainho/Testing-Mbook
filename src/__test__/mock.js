export const mockRateItems = [
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Trần Hùng",
    time: "31 phút trước",
    book: "../assets/book1.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "32 phút trước",
    book: "../assets/book4.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "33 phút trước",
    book: "../assets/book2.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "330 phút trước",
    book: "../assets/book3.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar1.jpg",
    author_name: "Hoài Nhớ",
    time: "301 phút trước",
    book: "../assets/book1.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "30 phút trước",
    book: "../assets/book1.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "30 phút trước",
    book: "../assets/book1.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
  {
    avatar: "../assets/avatar.jpg",
    author_name: "Hoài Nhớ",
    time: "30 phút trước",
    book: "../assets/book1.jpg",
    title: "Anh Da Đen và những câu chuyện",
    content: `that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.`,
  },
];

export const mockDataUsers = [
  {
    case: "Đăng nhập với tài khoản và mật khẩu đúng như yêu cầu",
    username: "hung2",
    password: "hung2",
    result: "Login successfully",
  },
  {
    case: "Bỏ trống các trường không nhập giá trị",
    username: "",
    password: "",
    result: "Username or Password INVALID",
  },
  {
    case: "Nhập tài khoản nhưng không nhập mật khẩu",
    username: "hung2",
    password: "",
    result: "Username or Password INVALID",
  },
  {
    case: "Nhập mật khẩu nhưng không nhập tài khoản",
    username: "",
    password: "hung2",
    result: "Username or Password INVALID",
  },
  {
    case: `Nhập giá trị “Khoảng trắng”`,
    username: "    ",
    password: "   ",
    result: "Username or Password INVALID",
  },
  {
    case: `Nhập các giá trị có bao gồm khoảng trắng`,
    username: "hung hung",
    password: "hung hung",
    result: "Username or Password INVALID",
  },
  {
    case: `Nhập các giá trị gồm 1 2 kí tự`,
    username: "a",
    password: "a",
    result: "Username or Password INVALID",
  },
  {
    case: "Nhập các giá trị có bao gồm ký tự đặc biệt",
    username: "!32$%@  **",
    password: "()1#~~4)",
    result: "Username or Password INVALID",
  },
];
