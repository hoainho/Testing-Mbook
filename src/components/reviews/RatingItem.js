import React, { useState } from 'react'
import { Link } from 'react-router-dom'
import classnames from 'classname';
export default function RatingItem(props) {
    const {avatar= '', author_name= '',time= '', book= '',title= '',content= ''} = props
    const [like, setLike] = useState(false);
    const handleLikeReply = () => {
        let newValue = like;
        newValue = !newValue;
        setLike(newValue);
    }
    return (
        <div className="action__list rating">
            <div className="action__list-avatar rating__avatar">
                <img className="action__list-avatar-image" src={avatar} />
            </div>
            <div className="action__list-comments rating__content">
                <div >
                    <h3 className="action__list-comments-name rating__content-title">{author_name}</h3>
                    <span className="rating__content-date">
                        <i class="fa fa-clock-o pr-2" aria-hidden="true"></i> {time}
                    </span>
                </div>
                <div className="rating__content-container">
                    <div className="rating__content-container-image">
                        <img className="rating__content-container-image-pic" src={book} alt="image" />
                    </div>
                    <div>
                        <div className="rating__content-container-decsription">
                            <h2 className="rating__content-container-decsription-title">
                                {title}
                            </h2>
                            <p className="rating__content-container-decsription-text">
                               {content}
                            </p>
                            {/* <Link to="/reviewsDetails"> Đọc tiếp</Link> */}
                            {/* <a href="/reviewsDetails"> Đọc tiếp</a> */}
                        </div>
                    </div>
                </div>

            </div>
        </div>

    )
}
