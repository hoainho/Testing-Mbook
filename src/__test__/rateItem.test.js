import React from 'react'
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react'
import RatingItem from '../components/reviews/RatingItem';
import { mockRateItems } from './mock';


describe('Snapshoot testing react componsent', () => {
  const getRateItems = jest.fn(()=>mockRateItems)

  const data = getRateItems()[0]
  it('So sánh 1 component', () => {
    const tree = renderer.create(< RatingItem 
        avatar={data.avatar} 
        author_name={data.author_name} 
        book={data.book} 
        time={data.time} 
        content={data.content}  
        title={data.title}
      />).toJSON();
    expect(tree).toMatchSnapshot(); 
  });

})

describe("Snapshoot testing react componsents", () => {
  const getRateItems = jest.fn(()=>mockRateItems)
  const data = getRateItems()
  data.map((elm, index) => {
    it(`So sánh component thứ ${index}`, () => {
      const tree = renderer.create(< RatingItem 
          avatar={elm.avatar} 
          author_name={elm.author_name} 
          book={elm.book} 
          time={elm.time} 
          content={elm.content}  
          title={elm.title}
        />).toJSON();
      expect(tree).toMatchSnapshot(); 
    });
  })
});