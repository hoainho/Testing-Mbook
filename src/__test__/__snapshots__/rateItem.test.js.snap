// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshoot testing react componsent So sánh 1 component 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Trần Hùng
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        31 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsent So sánh sự thay đổi của 1 component khi thay 1 đường dẫn  1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Trần Hùng
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        31 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 0 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Trần Hùng
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        31 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 1 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        32 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book4.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 2 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        33 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book2.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 3 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        330 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book3.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 4 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar1.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        301 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 5 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        30 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 6 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        30 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshoot testing react componsents So sánh component thứ 7 1`] = `
<div
  className="action__list rating"
>
  <div
    className="action__list-avatar rating__avatar"
  >
    <img
      className="action__list-avatar-image"
      src="../assets/avatar.jpg"
    />
  </div>
  <div
    className="action__list-comments rating__content"
  >
    <div>
      <h3
        className="action__list-comments-name rating__content-title"
      >
        Hoài Nhớ
      </h3>
      <span
        className="rating__content-date"
      >
        <i
          aria-hidden="true"
          class="fa fa-clock-o pr-2"
        />
         
        30 phút trước
      </span>
    </div>
    <div
      className="rating__content-container"
    >
      <div
        className="rating__content-container-image"
      >
        <img
          alt="image"
          className="rating__content-container-image-pic"
          src="../assets/book1.jpg"
        />
      </div>
      <div>
        <div
          className="rating__content-container-decsription"
        >
          <h2
            className="rating__content-container-decsription-title"
          >
            Anh Da Đen và những câu chuyện
          </h2>
          <p
            className="rating__content-container-decsription-text"
          >
            that have a near-identical initial appearance for single, multiple, and disabled
      select elements across the top browsers. A few properties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
      perties and techniques our
      solution will use: clip-path to create the custom dropdown arrow.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
